lib/v2/shmtu/portal.js

```diff
@@ -0,0 +1,61 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+const timezone = require('@/utils/timezone');
+const bootstrapHost = 'https://weixin.shmtu.edu.cn/dynamic/shmtuHttps';
+const host = 'https://portal.shmtu.edu.cn/api';
+
+const loadDetail = async (link) => {
+    const response = await got.post(bootstrapHost, {
+        form: {
+            interfaceUrl: link,
+        },
+        https: { rejectUnauthorized: false },
+    });
+
+    return JSON.parse(response.data);
+};
+
+const processFeed = (list, caches) =>
+    Promise.all(
+        list.map((item) =>
+            caches.tryGet(item.link, async () => {
+                const detail = await loadDetail(item.link);
+                item.description = detail.body.und[0].safe_value;
+                item.link = detail.path;
+                return item;
+            })
+        )
+    );
+
+module.exports = async (ctx) => {
+    const type = ctx.params.type;
+    const info = type === 'bmtzgg' ? '部门通知公告' : '未知';
+
+    const response = await got.post(bootstrapHost, {
+        form: {
+            interfaceUrl: `${host}/${type}.json?page=0`,
+        },
+        https: { rejectUnauthorized: false },
+    });
+
+    const list = JSON.parse(response.data).map((item) => ({
+        title: cheerio.load(item.title).text(),
+        link: `${host}/node/${item.nid}.json`,
+        pubDate: timezone(parseDate(item.created), 8),
+        category: item.field_department[0],
+        author: item.field_department[0],
+    }));
+
+    // 只取前五条
+    list.length = 5;
```

Comment:
Any reasons why slicing it to the first 5 items only?
There're sticky items in `response.data`. If there're more than 5 sticky items, the route can't parse new items that aren't sticky.

Reply:
According to current observations, a maximum of three news items are updated a day, and this list outputs 20 records by default, which may lead to failure of the PR bot tests (even though my local test is OK)